### PR TITLE
Refactor useSubjectImage as a shared hook

### PR DESCRIPTION
**Pull request by eatyourgreens**
- move `useSubjectImage` into `/src/hooks` and add some documentation.
- refactor the hook to handle errors and load events.
- update the flipbook viewer, multi-frame viewer and single image viewer to use the new hook.

## Package
lib-classifier

## How to Review
I've been testing this by checking the stories in the classifier storybook for the flipbook viewer, the multi-frame viewer and the single image viewer. Nothing should change with this refactor. Tested behaviour isn't changed here so the tests should continue to pass without changes.

This is a companion to #4285, in that the data-image viewer can use the new hook after this PR is merged.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Keeping this as draft until tests are passing.

**Comment by eatyourgreens** [comment]
Tests are passing now so this is ready.

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js [hunk 1]
```diff
@@ -1,62 +1,48 @@
...
-    if (src !== PLACEHOLDER_URL ) {
-      const svgImage = subjectImage.current
-      const { width: clientWidth, height: clientHeight } = svgImage
-        ? svgImage.getBoundingClientRect()
-        : {}
-      const target = { clientHeight, clientWidth, naturalHeight, naturalWidth }
-      onReady({ target })
-    }
-  }, [src])
+  const { img, error, loading, subjectImage } = useSubjectImage({
```
Comment (goplayoutside3) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js:36]: I noticed that in all three subject viewers, `error` and `loading` returned from `useSubjectImage()` aren’t used in the component. Do you have suggestions on how to utilize those variables? For instance, will there ever be a case when `img` is undefined?

**Review by goplayoutside3 — APPROVED** [review]
Looks good! Just a couple of non-blocking comments:

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js:36]: Those are there to catch errors. Maybe if the image URL 404s, perhaps?

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/FlipbookViewer/FlipbookViewer.js [hunk 2]
```diff
@@ -1,48 +1,50 @@
...
 const FlipbookViewer = ({
   defaultFrame = 0,
-  defaultFrameSrc = '',
-  enableRotation = () => true,
+  enableRotation = DEFAULT_HANDLER,
   flipbookAutoplay = false,
   invert = false,
   move,
-  naturalHeight = 600,
-  naturalWidth = 800,
```
[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/FlipbookViewer/FlipbookViewer.js:21]: Good point!

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js:36]: I think we can test this out in the storybook by adding a mock subject with a URL that errors. 

I put examples of how to use `loading` and `error` in the Readme too, but you’re right that the viewers aren’t using them.

**Pull request merged**